onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2,35 +2,32 @@
 // Licensed under the MIT License.
 
 #include <fstream>
-#include "core/graph/onnx_protobuf.h"
+#include <list>
+#include <unordered_set>
+#include "core/providers/shared_library/provider_api.h"
+#define ORT_API_MANUAL_INIT
+#include "core/session/onnxruntime_cxx_api.h"
+#include "core/common/safeint.h"
+
 #include "tensorrt_execution_provider.h"
 #include "core/providers/cuda/cuda_allocator.h"
+#include "core/providers/cuda/shared_inc/cuda_call.h"
 #include "core/providers/cuda/math/unary_elementwise_ops_impl.h"
-#include "core/session/onnxruntime_cxx_api.h"
-#include "core/framework/execution_provider.h"
-#include "core/framework/op_kernel.h"
-#include "core/framework/kernel_registry.h"
-#include "core/framework/compute_capability.h"
-#include "core/framework/memcpy.h"
-#include "core/providers/cpu/cpu_execution_provider.h"
-#include "core/providers/cuda/cuda_common.h"
-#include "core/providers/cuda/cuda_fence.h"
-#include "core/platform/env.h"
-#include "core/common/safeint.h"
-#include "core/common/status.h"
-#include "onnx/shape_inference/implementation.h"
 #include "cuda_runtime_api.h"
 #include "gsl/gsl"
-#include "core/graph/model.h"
-#include "core/providers/cuda/gpu_data_transfer.h"
 #include <experimental/filesystem>
 
+#define CUDA_RETURN_IF_ERROR(expr)               \
+  ORT_RETURN_IF_ERROR(CUDA_CALL(expr)            \
+                          ? common::Status::OK() \
+                          : ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "CUDA error executing ", #expr))
```

Comment:
This should be in common space

Reply:
Right now it's specific to only the tensorrt provider. Once I move cuda into the providers I'll have this with the other cuda things. It can't really be moved anywhere common as it depends on CUDA defines to work.